sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -94,7 +94,7 @@ public boolean accepts(Class<?> extensionType) {
                         invocationHandler));
     }
 
-    private Map<Method, Handler> methodHandlersFor(Class<?> sqlObjectType) {
+    private Map<Method, Handler> methodHandlersFor(Class<?> sqlObjectType, ConfigRegistry instanceConfig) {
```

Comment:
The handlers generated and cached in this method are class-level--they have no instance-specific data and can thus be safely reused by all instances of a given SQL object type.

Passing `instanceConfig` to this method mixes class- and instance-level scope--with the effect that config is locked in the first time a SQL object type is attached.

Moreover, it subverts a design intention in v3, that any changes made to a config class should be honored on a forward basis by new SQL object instances.

A couple options:

1. Leave it the way it is in this PR. Document that the default method handler config property must be set at the `Jdbi` level before the first SQL object is created.
2. Defer creation of the handler for a given method until the first time it is invoked. I rather dislike this approach, as it subverts our ability to sanity-check interfaces on `attach`. Also it adds some kind of bastard "meta-handler-factory" type which is kinda gross.
3. Make the default handler a property of `SqlObjectFactory` itself. This is a departure from the usual approach of just adding another config property, but given the pervasive effect of this setting I think it's justifiable. Apps (and plugins) could alter the default behavior as follows:

```java
jdbi.findFactory(SqlObjectFactory.java).ifPresent(f -> f.setDefaultMethodHandler(myMethodHandler));
```

@jdbi/contributors I'm leaving toward options 1 or 3. Opinions?